src/runlib.rs

```diff
@@ -1 +1,156 @@
 //! A tool that functionaries can use to create link metadata about a step.
+
+use std::collections::BTreeMap;
+use std::fs::{metadata, File};
+use std::io::{self, BufReader, Write};
+use std::process::Command;
+use walkdir::WalkDir;
+
+use crate::models::{Link, TargetDescription};
+use crate::Result;
+use crate::{
+    crypto,
+    crypto::PrivateKey,
+    models::{LinkMetadataBuilder, VirtualTargetPath},
+};
+
+/// recordArtifacts is a function that traverses through the passed slice of paths and logs the path
+/// and hashed file contents.
+pub fn record_artifacts(
+    paths: &[&str],
+    // hash_algorithms: Option<&[&str]>,
+) -> Result<BTreeMap<VirtualTargetPath, TargetDescription>> {
+    // TODO: if a step in record_artifact fails, and error should be returned instead of panicking
```

Comment:
Next step is definitely error handling here instead of unwrapping in various places, as well as handling edge cases (e.g. symbolic links).